tests/copy_test.go

```diff
@@ -0,0 +1,278 @@
+		&buildFileInfo{path: "dir01", typeflag: tar.TypeDir, mode: 0755, atime: time1, mtime: time1},
+		&buildFileInfo{path: "dir01/file01", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "hello"},
+	}
+
+	err = buildFS(sourceDir, files)
+	if err != nil {
+		panic(err)
+	}
+
+	err1 := runACBuild(workingDir, "copy", sourceDir, dest)
+	if err1 != nil {
+		t.Fatalf("%v\n", err1)
+	}
+
+	testMatchingFSTree(t, workingDir, sourceDir, dest)
+}
+
+func TestCopyOneFile(t *testing.T) {
+	workingDir := setUpTest(t)
+	defer cleanUpTest(workingDir)
+
+	sourceFile, err := ioutil.TempFile("", "acbuild-test-copy")
+	if err != nil {
+		panic(err)
+	}
+	_, err = sourceFile.Write([]byte("this is a test file"))
+	if err != nil {
+		panic(err)
+	}
+	sourceFile.Close()
+	defer os.RemoveAll(sourceFile.Name())
+
+	err1 := runACBuild(workingDir, "copy", sourceFile.Name(), dest)
+	if err1 != nil {
+		t.Fatalf("%v\n", err1)
+	}
+
+	testMatchingFSTree(t, workingDir, sourceFile.Name(), dest)
+}
+
+func TestCopyOneDirToExistingDir(t *testing.T) {
+	workingDir := setUpTest(t)
+	defer cleanUpTest(workingDir)
+
+	sourceDir, err := ioutil.TempDir("", "acbuild-test-copy")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(sourceDir)
+
+	err = os.MkdirAll(path.Join(workingDir, ".acbuild", "currentaci", aci.RootfsDir, dest), 0755)
+	if err != nil {
+		panic(err)
+	}
+
+	err1 := runACBuild(workingDir, "copy", sourceDir, dest)
+	if err1 == nil {
+		t.Fatalf("got no error when destination directory exists, was expecting one")
+	}
+}
+
+func TestCopyManyDirs(t *testing.T) {
+	workingDir := setUpTest(t)
+	defer cleanUpTest(workingDir)
+
+	sourceDir, err := ioutil.TempDir("", "acbuild-test-copy")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(sourceDir)
+
+	time1 := time.Now()
+	//time2 := time.Now().Add(1 * time.Microsecond)
+	files := []*buildFileInfo{
+		&buildFileInfo{path: "dir01", typeflag: tar.TypeDir, mode: 0755, atime: time1, mtime: time1},
+		&buildFileInfo{path: "dir01/file01", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "aaaa"},
+		&buildFileInfo{path: "dir02", typeflag: tar.TypeDir, mode: 0755, atime: time1, mtime: time1},
+		&buildFileInfo{path: "dir02/file02", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "bbbb"},
+		&buildFileInfo{path: "dir02/dir02.1", typeflag: tar.TypeDir, mode: 0755, atime: time1, mtime: time1},
+		&buildFileInfo{path: "dir02/dir02.1/file02", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "cccc"},
+		&buildFileInfo{path: "dir03", typeflag: tar.TypeDir, mode: 0755, atime: time1, mtime: time1},
+		&buildFileInfo{path: "dir03/file03", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "dddd"},
+	}
+
+	err = buildFS(sourceDir, files)
+	if err != nil {
+		panic(err)
+	}
+
+	froms := []string{"dir01", "dir02", "dir03"}
+	for i := 0; i < len(froms); i++ {
+		froms[i] = path.Join(sourceDir, froms[i])
+	}
+
+	// golang--
```

Comment:
And the type returned by runACBuild used to be a struct that implemented the error interface, and I was getting some weirdness at some point with that. That's not the case anymore, so this should go back to using `err`. My bad.